warehouse/manage/views/__init__.py

```diff
@@ -1158,8 +1164,178 @@ def manage_project_settings(self):
                     organization_choices=organization_choices,
                 )
             ),
+            "add_alternate_repository_form_class": add_alt_repo_form,
         }
 
+    @view_config(
+        request_method="POST",
+        request_param=AddAlternateRepositoryForm.__params__
+        + ["alternate_repository_location=add"],
+        require_reauth=True,
+        permission=Permissions.ProjectsWrite,
+    )
+    def add_project_alternate_repository(self):
+        form = self.add_alternate_repository_form_class(self.request.POST)
+
+        if not form.validate():
+            self.request.session.flash(
+                self.request._("Invalid alternate repository location details"),
+                queue="error",
+            )
+            return HTTPSeeOther(
+                self.request.route_path(
+                    "manage.project.settings",
+                    project_name=self.project.name,
+                )
+            )
+
+        # add the alternate repository location entry
+        alt_repo = AlternateRepository(
+            project=self.project,
+            name=form.display_name.data,
+            url=form.link_url.data,
+            description=form.description.data,
+        )
+        self.request.db.add(alt_repo)
+        self.request.db.add(
```

Comment:
Ope, totally missed this in review, but this is pretty incorrect and the cause of https://github.com/pypi/warehouse/issues/12933#issuecomment-2378120802